code/modules/mob/living/brain/MMI.dm

```diff
@@ -161,6 +161,7 @@
 /obj/item/mmi/proc/set_brainmob(mob/living/brain/new_brainmob)
 	if(brainmob == new_brainmob)
 		return FALSE
+	SEND_SIGNAL(src, COMSIG_MMI_SET_BRAINMOB, new_brainmob)
 	. = brainmob
```

Comment:
```suggestion
	. = brainmob
	SEND_SIGNAL(src, COMSIG_MMI_SET_BRAINMOB, new_brainmob)
```
Better send the signal after the parent return assignment, in case a response to the signal is added in the future.